db/version_set.cc

```diff
@@ -4303,11 +4307,41 @@ Status VersionSet::ProcessManifestWrites(
     for (auto v : versions) {
       delete v;
     }
+    if (manifest_io_status.ok()) {
+      manifest_file_number_ = pending_manifest_file_number_;
+      manifest_file_size_ = new_manifest_file_size;
+    }
     // If manifest append failed for whatever reason, the file could be
     // corrupted. So we need to force the next version update to start a
     // new manifest file.
     descriptor_log_.reset();
-    if (new_descriptor_log) {
+    // If manifest operations failed, then we know the CURRENT file still
```

Comment:
If `rename()` failed, should it just keep retrying? I guess we will retry `ProcessManifestWrites()` later by re-writing the manifest file and call `rename()` again, but should it just retry the single failed operation `rename()`?

Reply:
We do not have a good retry policy, so it is better to return the error and let application decide. We cannot keep retrying because we do not know how long it takes.
In the meantime, RocksDB should make sure there is no data loss. As you mentioned, the first subsequent `LogAndApply()` will do retry as necessary.